src/Microsoft.AspNetCore.Mvc.ViewFeatures/Rendering/IHtmlHelper.cs

```diff
@@ -234,6 +234,46 @@ public interface IHtmlHelper
             string htmlFieldName,
             object additionalViewData);
 
+        /// <summary>
+        /// Returns HTML markup for the <paramref name="expression"/>, using a display template, specified HTML field
+        /// name, and additional view data. The template is found using the <paramref name="templateName"/> or the
+        /// <paramref name="expression"/>'s <see cref="ModelMetadata"/>.
+        /// </summary>
+        /// <param name="expression">
+        /// Expression name, relative to the current model. May identify a single property or an
+        /// <see cref="object"/> that contains the properties to display.
+        /// </param>
+        /// <param name="templateName">The name of the template used to create the HTML markup.</param>
+        /// <param name="htmlFieldName">
+        /// A <see cref="string"/> used to disambiguate the names of HTML elements that are created for
+        /// properties that have the same name.
+        /// </param>
+        /// <param name="additionalViewData">
+        /// An anonymous <see cref="object"/> or <see cref="IDictionary{String, Object}"/> that can contain additional
+        /// view data that will be merged into the <see cref="ViewDataDictionary{TModel}"/> instance created for the
+        /// template.
+        /// </param>
+        /// <returns>A new <see cref="IHtmlContent"/> containing the created HTML.</returns>
+        /// <remarks>
+        /// <para>
+        /// For example the default <see cref="object"/> display template includes markup for each property in the
+        /// <paramref name="expression"/>'s value.
+        /// </para>
+        /// <para>
+        /// Example <paramref name="expression"/>s include <c>string.Empty</c> which identifies the current model and
+        /// <c>"prop"</c> which identifies the current model's "prop" property.
+        /// </para>
+        /// <para>
+        /// Custom templates are found under a <c>DisplayTemplates</c> folder. The folder name is case-sensitive on
+        /// case-sensitive file systems.
+        /// </para>
+        /// </remarks>
+        Task<IHtmlContent> DisplayAsync(
```

Comment:
Unfortunately any changes to an interface would be a breaking change that we cannot make in 2.2

Reply:
Oh, that's very sad.
Even though it's adding new methods and not modifying any existing methods?

Reply:
If this couldn't be merged,
Perhaps only the BuildAsync method of TemplateBuilder could be merged?
This would allow creating a custom extension methods similar to EditorFor and DisplayFor, using the sample TemplateBuilder.
This doesn't require changing any interface :)

Reply:
Yup. https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md#source-and-binary-compatibility-changes has a guideline for this. Adding a member to an interface is right up there.

Reply:
OK, guess I'll wait for the 3.0 release,
In the meanwhile I switched to a local build with these changes.
Anyone suffering from https://github.com/aspnet/KestrelHttpServer/issues/2375 should consider going for a local build, as it should improve their stability.
I'll update once we go live, seeing if this actually solved our issue.

Reply:
Reducing this PR to just the `TemplateBuilder` and `TemplateRenderer` changes would be reasonable in 2.2. However, we wouldn't recommend using either outside this assembly: Both are in the Microsoft.AspNetCore.Mvc.ViewFeatures.**Internal** namespace.